finality-aleph/src/data_io/data_store.rs

```diff
@@ -117,8 +117,8 @@ impl Default for DataStoreConfig {
             max_proposals_pending: 80_000,
             max_messages_pending: 40_000,
             available_proposals_cache_capacity: 8000,
-            periodic_maintenance_interval: Duration::from_secs(60),
-            request_block_after: Duration::from_secs(100),
+            periodic_maintenance_interval: Duration::from_secs(25),
+            request_block_after: Duration::from_secs(20),
```

Comment:
In every maintenance run we always touch every proposal, but the `request_block_after` determines how much time must pass from the point in time when we first received a given proposal till when we first attempt requesting a block or justification. This means that after we receive a proposal and it stays pending, then either in the first or the second maintenance, something will be requested. I added a comment explaining this parameter.

In general these delays are probably quite large, but we are really stepping on thin ice here, because there is uncertainty on how well the network code handles multiple requests in short periods of time (i.e. whether it will lead to mass banning of nodes because of some random thing).

Reply:
And this stuff is virtually impossible to test properly :/